Source/SelfService/Web/microservice/dataTable.tsx

```diff
@@ -0,0 +1,211 @@
+
+import { getPodStatus, MicroserviceInfo } from '../api/api';
+import { HttpResponseApplication } from '../api/application';
+
+import { DataGridPro, GridColDef, GridValueGetterParams, GridRenderCellParams } from '@mui/x-data-grid-pro';
+
+import { themeDark } from '../theme/theme';
+import { Box, Tooltip, Typography } from '@mui/material';
+import { CheckCircleRounded, ErrorRounded, WarningRounded } from '@mui/icons-material';
+
+const styles = {
+    dataTable: {
+        'background': 'linear-gradient(180deg, rgba(255, 255, 255, 0.05) 0%, rgba(255, 255, 255, 0.05) 100%), #191A21;',
+        '.MuiDataGrid-row': {
+            cursor: 'pointer'
+        },
+        '.MuiDataGrid-columnSeparator': {
+            display: 'none'
+        },
+        '.MuiDataGrid-columnHeader:focus': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-columnHeader:focus-within': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-columnHeaderDraggableContainer': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-cell:focus-within': {
+            outline: 'none'
+        },
+        '.MuiIconButton-root': {
+            'visibility': 'visible',
+            ':hover': {
+                backgroundColor: 'transparent'
+            }
+        },
+        '.MuiDataGrid-sortIcon': {
+            color: 'rgba(255, 255, 255, 0.38);',
+            width: '20px',
+            height: '20px'
+        },
+        '.MuiDataGrid-columnHeader:not(.MuiDataGrid-columnHeader--sorted) .MuiDataGrid-sortIcon': {
+            opacity: 1
+        }
+    },
+    status: {
+        display: 'flex',
+        justifyContent: 'center'
+    },
+    statusTitle: {
+        fontWeight: '500',
+        fontSize: '12px',
+        lineHeight: '22px',
+        letterSpacing: '0.06em',
+        textTransform: 'uppercase',
+        marginInlineStart: '10px'
+    }
+};
+
+export type MicroserviceObject = {
+    id: string,
+    name: string,
+    kind: string,
+    environment: string,
+    live: MicroserviceInfo,
+    edit: {},
+    status?: []
+};
+
+type DataTableProps = {
+    environment: string,
+    application: HttpResponseApplication,
+    microservices: MicroserviceObject[]
+};
+
+export const DataTable: React.FC<DataTableProps> = ({ application, environment, microservices }: DataTableProps) => {
+    const history = useHistory();
+    const [rows, setRows] = useState<(MicroserviceObject | undefined)[]>([]);
+
+    useEffect(() => {
+        const mapStatusToMicroservicesAndSetState = async () => {
+            const rows = await Promise.all(microservices.map(async microservice => {
+                const status = await getMicroserviceStatus(microservice.id);
+
+                return {
+                    ...microservice,
+                    status
+                } as MicroserviceObject;
+            }));
+
+            setRows(rows);
+        };
+
+        mapStatusToMicroservicesAndSetState();
+    }, [setRows]);
```

Comment:
This should include microservices as a dependency:
```suggestion
    }, [setRows, microservices]);
```